File: src/screens/LoginScreen.js
import React, { useState, useEffect } from 'react';
import { View, Text, TouchableOpacity, StyleSheet, SafeAreaView } from 'react-native';
import { useTheme } from '../context/ThemeContext';
import { useAuth } from '../context/AuthContext';
import { Logo } from '../components/Logo';

const PHRASES = [
  'growthBox',
  'Grow Daily',
  'Level Up',
  'Next Step',
  'Move Forward',
  'Unlock Potential',
];

export const LoginScreen = () => {
  const { theme } = useTheme();
  const { loginWithLocalStorage } = useAuth();

  const [phraseIndex, setPhraseIndex] = useState(0);
  const [typedText, setTypedText] = useState('');
  const [isDeleting, setIsDeleting] = useState(false);

  useEffect(() => {
    const timer = setTimeout(() => {
      const currentPhrase = PHRASES[phraseIndex];
      if (!isDeleting) {
        if (typedText.length < currentPhrase.length) {
          setTypedText(currentPhrase.slice(0, typedText.length + 1));
        } else {
          setTimeout(() => setIsDeleting(true), 1500);
        }
      } else {
        if (typedText.length > 0) {
          setTypedText(currentPhrase.slice(0, typedText.length - 1));
        } else {
          setIsDeleting(false);
          setPhraseIndex((prev) => (prev + 1) % PHRASES.length);
        }
      }
    }, isDeleting ? 45 : 95);

    return () => clearTimeout(timer);
  }, [typedText, isDeleting, phraseIndex]);

  return (
    <SafeAreaView style={[styles.container, { backgroundColor: theme.bgPrimary }]}>
      <View style={styles.center}>
        <Logo size={76} showText={false} />
        <View style={styles.sloganRow}>
          <Text style={[styles.sloganText, { color: theme.textPrimary }]}>{typedText}</Text>
          <Text style={styles.cursor}>|</Text>
        </View>
        <Text style={[styles.subtitle, { color: theme.textSecondary }]}>
          Manage your notes and todos with AI.{'\n'}Simple, fast, and intelligent.
        </Text>

        <TouchableOpacity
          style={styles.localBtn}
          onPress={loginWithLocalStorage}
          activeOpacity={0.8}
        >
          <Text style={styles.localBtnText}>⚡ Continue with localStorage (100% Offline)</Text>
        </TouchableOpacity>

        <TouchableOpacity
          style={[styles.outBtn, { borderColor: theme.borderPrimary }]}
          onPress={() => alert('Online authentication is disabled in offline mode. Please use localStorage.')}
        >
          <Text style={[styles.outText, { color: theme.textSecondary }]}>
            Continue with Google (Cloud Sync)
          </Text>
        </TouchableOpacity>

        <TouchableOpacity
          style={[styles.outBtn, { borderColor: theme.borderPrimary }]}
          onPress={() => alert('Online authentication is disabled in offline mode. Please use localStorage.')}
        >
          <Text style={[styles.outText, { color: theme.textSecondary }]}>
            Continue with GitHub (Cloud Sync)
          </Text>
        </TouchableOpacity>

        <Text style={[styles.footerText, { color: theme.textTertiary }]}>
          growthBox React Native v1.0 • 100% Offline Capable & Privacy-First
        </Text>
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  center: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    paddingHorizontal: 28,
  },
  sloganRow: {
    flexDirection: 'row',
    alignItems: 'center',
    marginTop: 24,
  },
  sloganText: {
    fontSize: 32,
    fontWeight: '800',
  },
  cursor: {
    fontSize: 32,
    fontWeight: '300',
    color: '#FCBB00',
    marginLeft: 2,
  },
  subtitle: {
    textAlign: 'center',
    fontSize: 14,
    lineHeight: 20,
    marginTop: 10,
    marginBottom: 36,
  },
  localBtn: {
    width: '100%',
    paddingVertical: 15,
    borderRadius: 14,
    backgroundColor: '#FCBB00',
    alignItems: 'center',
    marginBottom: 12,
  },
  localBtnText: {
    color: '#000',
    fontSize: 14,
    fontWeight: '700',
  },
  outBtn: {
    width: '100%',
    paddingVertical: 14,
    borderRadius: 14,
    borderWidth: 1,
    alignItems: 'center',
    marginBottom: 10,
  },
  outText: {
    fontSize: 13,
    fontWeight: '500',
  },
  footerText: {
    fontSize: 11,
    marginTop: 32,
  },
});

File: lib/screens/login_screen.dart
import 'dart:async';
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';
import '../providers/auth_provider.dart';
import '../theme/colors.dart';
import '../widgets/logo_widget.dart';
import 'home_screen.dart';

class LoginScreen extends StatefulWidget {
  const LoginScreen({super.key});

  @override
  State<LoginScreen> createState() => _LoginScreenState();
}

class _LoginScreenState extends State<LoginScreen> {
  final List<String> _phrases = [
    'growthBox',
    'Grow Daily',
    'Level Up',
    'Next Step',
    'Move Forward',
    'Unlock Potential',
  ];

  int _phraseIndex = 0;
  String _currentText = '';
  int _charIndex = 0;
  bool _isDeleting = false;
  Timer? _timer;

  @override
  void initState() {
    super.initState();
    _startTypingAnimation();
  }

  void _startTypingAnimation() {
    _timer = Timer.periodic(const Duration(milliseconds: 90), (timer) {
      if (!mounted) return;
      final targetWord = _phrases[_phraseIndex];

      setState(() {
        if (!_isDeleting) {
          if (_charIndex < targetWord.length) {
            _charIndex++;
            _currentText = targetWord.substring(0, _charIndex);
          } else {
            _isDeleting = true;
            _timer?.cancel();
            Future.delayed(const Duration(milliseconds: 1800), () {
              if (mounted) _startTypingAnimation();
            });
          }
        } else {
          if (_charIndex > 0) {
            _charIndex--;
            _currentText = targetWord.substring(0, _charIndex);
          } else {
            _isDeleting = false;
            _phraseIndex = (_phraseIndex + 1) % _phrases.length;
          }
        }
      });
    });
  }

  @override
  void dispose() {
    _timer?.cancel();
    super.dispose();
  }

  Future<void> _handleLocalLogin() async {
    final authProvider = Provider.of<AuthProvider>(context, listen: false);
    await authProvider.loginWithLocalStorage();
    if (!mounted) return;
    Navigator.of(context).pushReplacement(
      MaterialPageRoute(builder: (_) => const HomeScreen()),
    );
  }

  void _showOnlineLoginToast(String provider) {
    ScaffoldMessenger.of(context).showSnackBar(
      SnackBar(
        content: Text('Continue with $provider via maniweb API'),
        duration: const Duration(seconds: 2),
      ),
    );
  }

  @override
  Widget build(BuildContext context) {
    final theme = Theme.of(context);
    final isDark = theme.brightness == Brightness.dark;

    return Scaffold(
      body: SafeArea(
        child: Center(
          child: SingleChildScrollView(
            padding: const EdgeInsets.symmetric(horizontal: 32, vertical: 40),
            child: Column(
              mainAxisAlignment: MainAxisAlignment.center,
              children: [
                const LogoWidget(size: 86, showText: false),
                const SizedBox(height: 32),
                Row(
                  mainAxisAlignment: MainAxisAlignment.center,
                  children: [
                    Text(
                      _currentText,
                      style: theme.textTheme.displayLarge?.copyWith(
                        fontSize: 34,
                        fontWeight: FontWeight.w800,
                        letterSpacing: -1.0,
                      ),
                    ),
                    Text(
                      '|',
                      style: TextStyle(
                        fontSize: 34,
                        fontWeight: FontWeight.w300,
                        color: theme.primaryColor,
                      ),
                    ),
                  ],
                ),
                const SizedBox(height: 16),
                Text(
                  'Manage your notes and todos with AI.\nSimple, fast, and intelligent.',
                  textAlign: TextAlign.center,
                  style: theme.textTheme.bodyMedium?.copyWith(
                    fontSize: 15,
                    height: 1.5,
                  ),
                ),
                const SizedBox(height: 48),

                // Continue with localStorage button (100% Offline)
                SizedBox(
                  width: double.infinity,
                  height: 54,
                  child: ElevatedButton.icon(
                    onPressed: _handleLocalLogin,
                    icon: const Icon(Icons.offline_bolt, size: 22),
                    style: ElevatedButton.styleFrom(
                      backgroundColor: AppColors.amber400,
                      foregroundColor: Colors.black,
                      elevation: 0,
                      shape: RoundedRectangleBorder(
                        borderRadius: BorderRadius.circular(12),
                      ),
                    ),
                    label: const Text(
                      'Continue with localStorage (100% Offline)',
                      style: TextStyle(
                        fontSize: 15,
                        fontWeight: FontWeight.w700,
                      ),
                    ),
                  ),
                ),
                const SizedBox(height: 14),

                // Continue with Google button
                SizedBox(
                  width: double.infinity,
                  height: 50,
                  child: OutlinedButton.icon(
                    onPressed: () => _showOnlineLoginToast('Google'),
                    icon: const Icon(Icons.g_mobiledata, size: 26),
                    label: const Text(
                      'Continue with Google (Cloud Sync)',
                      style: TextStyle(fontSize: 14, fontWeight: FontWeight.w500),
                    ),
                    style: OutlinedButton.styleFrom(
                      foregroundColor: theme.textTheme.titleLarge?.color,
                      side: BorderSide(color: theme.dividerColor.withOpacity(0.25)),
                      shape: RoundedRectangleBorder(
                        borderRadius: BorderRadius.circular(12),
                      ),
                    ),
                  ),
                ),
                const SizedBox(height: 12),

                // Continue with GitHub button
                SizedBox(
                  width: double.infinity,
                  height: 50,
                  child: OutlinedButton.icon(
                    onPressed: () => _showOnlineLoginToast('GitHub'),
                    icon: const Icon(Icons.code, size: 20),
                    label: const Text(
                      'Continue with GitHub (Cloud Sync)',
                      style: TextStyle(fontSize: 14, fontWeight: FontWeight.w500),
                    ),
                    style: OutlinedButton.styleFrom(
                      foregroundColor: theme.textTheme.titleLarge?.color,
                      side: BorderSide(color: theme.dividerColor.withOpacity(0.25)),
                      shape: RoundedRectangleBorder(
                        borderRadius: BorderRadius.circular(12),
                      ),
                    ),
                  ),
                ),
                const SizedBox(height: 36),
                Text(
                  'growthBox Android v1.0 • 100% Offline Capable & Privacy-First',
                  style: theme.textTheme.bodySmall?.copyWith(
                    fontSize: 11,
                    color: theme.textTheme.bodySmall?.color?.withOpacity(0.7),
                    fontWeight: FontWeight.w500,
                  ),
                ),
              ],
            ),
          ),
        ),
      ),
    );
  }
}
